src/dropdown/dropdown.ts

```diff
@@ -36,24 +36,38 @@ export class NgbDropdownMenu {
 
   isEventFrom($event) { return this._elementRef.nativeElement.contains($event.target); }
 
-  position(triggerEl, placement) {
-    this.applyPlacement(positionElements(triggerEl, this._elementRef.nativeElement, placement));
+  position(triggerEl, placement, container) {
+    this.applyPlacement(positionElements(triggerEl, this._elementRef.nativeElement, placement, container === 'body'));
+  }
+
+  applyContainer(container: string) {
+    const nativeElement = this._elementRef.nativeElement;
+    if (container === 'body') {
+      document.body.appendChild(nativeElement);
+    }
+  }
+
+  removeFromBody() {
+    const nativeElement = this._elementRef.nativeElement;
+    if (nativeElement.parentNode === document.body) {
+      document.body.removeChild(nativeElement);
```

Comment:
I don't think this is correct - we should rather return it back to the original parent. Think of this situation:
* we start with `[container]="exp"` where `exp` evaluates to `"body"`;
* we open and close dropdown;
* we flip `exp` to evaluate to `null`
* we try to open a dropdown - it is detached from the DOM now!

Reply:
Thanks for your comment. I'm going to have a look at this use case